sds/src/proximity_keywords.rs

```diff
@@ -127,6 +135,12 @@ impl CompiledProximityKeywords {
     }
 }
 
+fn contains_keyword_in_path(path: &str, regex: &ProximityKeywordsRegex<false>) -> bool {
```

Comment:
Another way to look at this is to use a more oriented object code. In that sense, you can implement `ProximityKeywordsRegex` with two methods: `contains_keywords_in_content` and `contains_keywords_in_path`. 

```rust
impl<const EXCLUDED_CHARS: bool> ProximityKeywordsRegex<EXCLUDED_CHARS> {
    fn contains_keywords_in_content(
        &self,
        content: &str,
        match_start: usize,
        look_ahead_char_count: usize,
    ) -> bool {
      ...
   }

   fn contains_keyword_in_path(&self, path: &str) -> bool {
       ...
   }
```

All the logic related to the match will be encapsulated in the `ProximityKeywordsRegex`. It will be less likely to use `contains_keywords_in_content` with the regex associated to the `path` and vice versa.

Reply:
[Move functions to methods to have more resilient code](https://github.com/DataDog/dd-sensitive-data-scanner/pull/53/commits/f30896ea6cefd294e7bc98bfb9d53810db397a81)